src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlDataReader.cs

```diff
@@ -834,11 +834,18 @@ private void CleanPartialReadReliable()
         /// <include file='../../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlDataReader.xml' path='docs/members[@name="SqlDataReader"]/Dispose/*' />
         protected override void Dispose(bool disposing)
         {
-            if (disposing)
+            try
             {
-                Close();
+                if (disposing)
+                {
+                    Close();
+                }
+                base.Dispose(disposing);
+            }
+            catch(SqlException ex)
+            {
+                SqlClientEventSource.Log.TryTraceEvent("SqlDataReader.Dispose | ERR | Error: {0}", ex);
```

Comment:
Do we want to this catching in Close instead? That way it will be in parity with .NET Framework, and that's the only place any driver errors will be caught.

Also make changes in .NET Framework (SqlDataReader.Close()) method.

Reply:
That's what his original change did. But we don't want to change the behavior or Close(), just Dispose(). Dispose() is what carries the best practice that it should not throw. Also, if we want parity with netfx, we would bring the Dispose() method with its try/catch to netfx and leave Close() as-is.

Reply:
I have added Dispose() with try/catch block to netfx for parity as @David-Engel suggested.